### PR TITLE
no_sharedtypedarrays

### PR DESCRIPTION
**Pull request by juj**
Remove references to old Shared*Arrays which are not part of the final spec. #3869.

This not yet good to merge! Only putting up for initial review.

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -1228,32 +1228,23 @@ if (typeof SharedArrayBuffer !== 'undefined') {
...
-  HEAP8 = new Int8Array(buffer);
-  HEAP16 = new Int16Array(buffer);
-  HEAP32 = new Int32Array(buffer);
-  HEAPU8 = new Uint8Array(buffer);
-  HEAPU16 = new Uint16Array(buffer);
-  HEAPU32 = new Uint32Array(buffer);
-  HEAPF32 = new Float32Array(buffer);
-  HEAPF64 = new Float64Array(buffer);
 }
+HEAP8 = new Int8Array(buffer);
```
Comment (kripken) [inline src/preamble.js]: this can now be replaced with a call to `updateGlobalBufferViews` (recently added in this file)

**Comment by kripken** [comment]
Nice!

**Comment by juj** [comment]
Looks like we need to wait until https://bugzilla.mozilla.org/show_bug.cgi?id=1176214 lands to land this.

File: src/preamble.js (hunk 1 above)
Comment (juj) [inline src/preamble.js]: Oh, looks good - updated two instances in `src/preamble.js` to use this.

**Comment by kripken** [comment]
lgtm, land when appropriate.

**Comment by kripken** [comment]
Landed this now, after seeing the update on nightly (which this unbreaks).

**Pull request merged**